archivy/data.py

```diff
@@ -314,3 +314,35 @@ def open_file(path):
         subprocess.Popen(["open", path])
     else:
         subprocess.Popen(["xdg-open", path])
+
+
+def valid_image_filename(filename):
+    ALLOWED_EXTENSIONS = ["jpg", "png", "gif"]
+    return "." in filename and filename.rsplit(".")[1] in ALLOWED_EXTENSIONS
+
+
+def save_image(image):
+    """
+    Saves image to USER_DATA_DIR
+
+    Returns: filename where image has been saved.
+    """
+    base_path = Path(current_app.config["USER_DIR"]) / "images"
+    fileparts = image.filename.rsplit(".", 1)
+    sanitized_filename = secure_filename(fileparts[0])
+    dest_path = base_path / f"{sanitized_filename}.{fileparts[1]}"
+    i = 1
+    while dest_path.exists():
+        dest_path = base_path / f"{sanitized_filename}-{i}.{fileparts[1]}"
+        i += 1
+    image.save(str(dest_path))
+    return dest_path.parts[-1]
+
+
+def image_exists(filename):
```

Comment:
Can we add a type hint over here, it looks like it can be both an `int` or a `str`.